signac/contrib/import_export.py

```diff
@@ -241,6 +241,26 @@ def path_function(job):
             except Exception as error:
                 raise _SchemaPathEvaluationError(error)
 
+        # Check that the user-specified path generates a 1-1 mapping
+        links = set()
+        duplicate_links = set()  # helps only log once per duplicate
+        for job in jobs:
+            job_path = path_function(job)
+            if job_path not in duplicate_links and job_path in links:
+                logger.debug(f"Generated path '{job_path}' is not unique.")
+                duplicate_links.add(job_path)
+            else:
+                links.add(job_path)
+        num_dups = len(duplicate_links)
+        if num_dups > 0:
+            path_correction = os.path.join("id", "{job.id}")
+            raise RuntimeError(
+                f"The path specification '{path}' would result in {num_dups} duplicate "
+                "links. See the debug log for the list. The easiest way to fix "
+                f"this is to append the job id to the path specification like '{path_correction}'."
+            )
+        logger.info(f"Path specification {path} uniquely maps workspaces.")
```

Comment:
In this implementation, I have a few comments. I am not sure if this is my favored implementation after reading Vyas's comment. See https://github.com/glotzerlab/signac/pull/642/#discussion_r752313976
1. I don't think `duplicate_links` is necessary to track. I would report all duplicates regardless of how many times they occur -- that feels more like canonical behavior for UNIX command-line software.
2. It is probably not necessary to report the number of duplicates. The user can't do much with that information.
3. I don't think it is necessary to log the info message on success.
```suggestion
        # Check that the user-specified path generates a 1-1 mapping
        links = set()
        for job in jobs:
            job_path = path_function(job)
            if job_path in links:
                logger.debug(f"Generated path '{job_path}' is not unique.")
            else:
                links.add(job_path)
        if len(links) != len(jobs):
            path_correction = 
            raise RuntimeError(
                f"The path specification '{path}' would result in duplicate "
                "links. See the debug log for the list. The easiest "
                "way to fix this is to append the job id to the path "
                f"specification like '{os.path.join("id", "{job.id}")}'."
            )
```